server/server.js

```diff
@@ -20,52 +16,25 @@ App.use(json());
 App.use(express.static(join(__dirname, "../public")));
 App.use(expressPino);
 
-App.get("/", (req, res) => res.sendFile(join(__dirname, "../public/index.html")));
-App.get("/healthz", (req, res) => {
-    runHealthChecks().then(result => {
-        res.status(result.ok ? 200 : 503);
-        return res.json(result);
-    }).catch(err => {
-        log.error(err);
-        return res.sendStatus(500);
-    });
-});
-
-setupGraphQLServer(App);
+App.use("/", router);
 
-App.get("/schema/:schemaId", async (req, res) => {
-    try {
-        const s = await schema.findById(req.params.schemaId);
-        if (s) {
-            const compiled = await compileSchemaString(s.schema);
-            const filename = `schema_${s.name}.json`;
-            // tell the browser to handle this as a file download
-            res.setHeader("Content-disposition", `attachment; filename=${filename}`);
-            res.setHeader("Content-type", "text/json");
-            res.status(200).send(JSON.stringify(compiled, null, 2));
-        } else {
-            res.sendStatus(404);
-        }
-    } catch (err) {
-        log.error(err);
-        res.sendStatus(500);
-    }
-});
-
-// Catch all other requests and return "Not found"
-App.get("*", (_, res) => res.sendStatus(404));
+let server = null;
 
+// FIXME: app start/close logic needs to wrap DB initialization, otherwise it is
+// not usable after calling stop().
 exports.run = callback => {
     sync().then(() => {
         server = App.listen(port, () => callback(App));
     });
 };
 
-exports.stop = () => {
+exports.stop = callback => {
     log.info("Shutting down UI server");
-    server.close();
     stopNotifications();
-    database.close();
+    database.close()
+        .then(() => {
+            server.close(() => callback());
```

Comment:
This throws and error: https://gist.github.com/psturc/76381f9156cfa899f977614fdf0f338d (after starting and stopping the server)